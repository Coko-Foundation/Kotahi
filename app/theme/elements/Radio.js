import { css, keyframes } from 'styled-components'
import { th } from '@pubsweet/ui-toolkit'

const checking = keyframes`
  0% {
    transform: scale(0.8);
  }
  20% {
    transform: scale(1.2);
  }
  80% {
    transform: scale(1);
  }
  100% {
    transform: scale(1);
  }
`
export default {
  Root: css`
    transition: all ${th('transitionDuration')};

    &:hover {
      span {
        color: ${props =>
          props.checked ? 'inherit' : props.theme.colorPrimary};

        &:before {
          animation-name: ${props => (props.checked ? 'none' : checking)};
          animation-duration: ${th('transitionDuration')};
          box-shadow: 0 0 0 ${th('borderWidth')}
            ${props =>
              props.checked ? 'currentColor' : props.theme.colorPrimary};
        }
      }
    }
  `,
  Label: css`
    font-style: italic;

    &:before {
      content: ' ';
      display: inline-block;
      vertical-align: middle;
      width: calc(${th('gridUnit')} * 2);
      height: calc(${th('gridUnit')} * 2);
      margin-left: ${th('gridUnit')};
      margin-right: ${th('gridUnit')};

      /* This is not a real border (box-shadow provides that), so not themed as such */
      border: calc(${th('gridUnit')} / 4) solid white;
      border-radius: 50%;

      transition: border ${th('transitionDuration')}
        ${th('transitionTimingFunction')};
      color: ${props => (props.color ? props.color : props.theme.colorText)};
      background: ${props => (props.checked ? 'currentColor' : 'transparent')};
      box-shadow: 0 0 0 ${th('borderWidth')} currentColor;
    }
  `,
  Input: css`
    position: absolute;
    opacity: 0;
    z-index: -1;

    &:focus + span:before {
      box-shadow: 0 0 ${th('borderWidth')} calc(${th('borderWidth')} * 2)
        ${th('colorPrimary')};
    }
  `,
}
