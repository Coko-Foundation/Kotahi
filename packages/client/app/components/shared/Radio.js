import React from 'react'
import styled, { css } from 'styled-components'
import { th, override } from '@pubsweet/ui-toolkit'

const Label = styled.span`
  ${props =>
    props.disabled &&
    css`
      cursor: not-allowed;
    `};

  ${override('ui.Label')};
  ${override('ui.Radio.Label')};
`

const Input = styled.input`
  margin-right: ${th('gridUnit')};
  ${override('ui.Radio.Input')};
`

const Root = styled.label`
  align-items: center;
  color: ${props => (props.color ? props.color : props.theme.colorText)};
  cursor: pointer;
  display: ${props => (props.inline ? 'inline-flex' : 'flex')};
  min-height: calc(${th('gridUnit')} * 3);

  ${override('ui.Radio')};
`

// eslint-disable-next-line import/prefer-default-export
export const Radio = ({
  className,
  disabled,
  color,
  inline,
  name,
  value,
  label,
  checked,
  required,
  onChange,
}) => (
  <Root checked={checked} className={className} color={color} inline={inline}>
    <Input
      checked={checked}
      disabled={disabled}
      name={name}
      onChange={onChange}
      required={required}
      type="radio"
      value={value}
    />
    <Label checked={checked} color={color} disabled={disabled}>
      {label}
    </Label>
  </Root>
)
