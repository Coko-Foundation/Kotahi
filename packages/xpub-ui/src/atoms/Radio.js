import React from 'react'
import classes from './Radio.local.css'

const Radio = ({ name, value, label, checked, required, handleChange }) => (
  <label className={classes.root}>
    <input
      className={classes.input}
      type="radio"
      name={name}
      value={value}
      checked={checked}
      required={required}
      onChange={event => handleChange(event.target.value)}/>
    {label}
  </label>
)

export default Radio
