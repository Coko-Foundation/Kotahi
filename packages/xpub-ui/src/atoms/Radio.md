A radio button.

```js
<Radio 
  name="foo" 
  handleChange={event => console.log(event)}/>
```

A checked radio button.

```js
<Radio 
  name="bar" 
  checked 
  handleChange={event => console.log(event)}/>
```

A radio button with a label.

```js
<Radio 
  name="foo" 
  label="Foo"
  handleChange={event => console.log(event)}/>
```
