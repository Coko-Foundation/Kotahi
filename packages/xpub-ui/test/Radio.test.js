import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import Radio from '../src/atoms/Radio'

const props = {
  checked: false,
  label: 'TestLabel',
  name: 'TestName',
  required: true,
  value: 'TestValue',
}

const wrapper = shallow(<Radio {...props} />)

describe('Radio', () => {
  test('Snapshot', () => {
    const tree = renderer.create(<Radio {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Input gets the correct props', () => {
    const input = wrapper.find('input')

    expect(input.prop('name')).toBe(props.name)
    expect(input.prop('value')).toBe(props.value)
    expect(input.prop('checked')).toBe(props.checked)
    expect(input.prop('required')).toBe(props.required)
  })
})
