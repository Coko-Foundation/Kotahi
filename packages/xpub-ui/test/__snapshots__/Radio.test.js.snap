// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Snapshot 1`] = `
<label
  className="root"
  style={
    Object {
      "color": "black",
    }
  }
>
  <input
    checked={false}
    className="input"
    name="TestName"
    onChange={undefined}
    required={true}
    type="radio"
    value="TestValue"
  />
  <span
    className="pseudoInput"
    style={
      Object {
        "background": "transparent",
      }
    }
  >
     
  </span>
  <span
    className="label"
  >
    TestLabel
  </span>
</label>
`;
